### ISSUE TITLE
SPU Compiler bug (regression?) after #5427 [BLES01294]

### ISSUE THREAD

**Issue by DumF0rGaming** [RPCS3/rpcs3#5553]
Red Dead Redemption v1.0 has been compiling just fine on build 7676 from here
https://www.youtube.com/watch?v=IZCXx5BGkCo

and now I'm experiencing an odd SPU compiler crash without a single  module being compiled
![set](https://user-images.githubusercontent.com/5430116/51122000-dfdc6700-1829-11e9-883d-1b2af4accd2c.PNG)
Switching TSX between forced and enabled as well as Mega and Giga SPU Block sizes doesn't make any difference.
If I start the game for the first time, when it doesn't try to precompile SPU cache yet - the game's passing the intro and crashes without reaching the menu.

Here are the logs
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/2755772/RPCS3.log)
[TTY.log](https://github.com/RPCS3/rpcs3/files/2755773/TTY.log)

RPCS3 v0.0.5-7727-c5026f71 Alpha
Windows 10 v1809
Intel(R) Core(TM) i7-6700K
16 GB RAM
GeForce GTX 1080
Driver 416.94

**Comment by Asinin3** [comment]
Have you tried using the Safe block size, instead of the WiP/Experimental Mega block size? (giga is experimental too)

**Comment by DumF0rGaming** [comment]
With Safe block size the game boots and act as expected untill the cut-scene starts. I can hear it sounds but the emu itself still shows a loading screen and that it's busy compliling shaders. Async shader compiler is enabled, I'll give it another try without forcing the game to run in fullscreen mode. I can't share logs this time, they're 95MB big

**Comment by DumF0rGaming** [comment]
Looks like this game is also Fullscreen agnostic but it doesn't cancel the fact that something got wrong with Mega and Giga block sizes that were working on beforementioned build

**Comment by legend800** [comment]
So you saying the game used to be able to use SPU Mega and now it can't?

**Comment by DumF0rGaming** [comment]
Yeah

**Comment by Xcedf** [comment]
@DumF0rGaming this issue is duplicate of https://github.com/RPCS3/rpcs3/issues/5257
the build 7676, you've talking about has some tweaks to make compiling work
